generators/csharp/codegen/src/context/convertExampleTypeReferenceToTypeReference.ts

```diff
@@ -0,0 +1,202 @@
+        case "primitive":
+            return convertExamplePrimitiveTypeToTypeReference(exampleTypeReference.shape);
+        case "container":
+            return convertExampleContainerTypeToTypeReference(exampleTypeReference.shape);
+        case "unknown":
+            return FernIr.TypeReference.unknown();
+        default:
+            assertNever(exampleTypeReference.shape);
+    }
+}
+
+function convertExampleNamedTypeToTypeReference(
+    exampleNamedType: FernIr.ExampleTypeReferenceShape.Named
+): FernIr.TypeReference.Named {
+    return FernIr.TypeReference.named({
+        name: exampleNamedType.typeName.name,
+        inline: false,
+        default: undefined,
+        fernFilepath: exampleNamedType.typeName.fernFilepath,
+        typeId: exampleNamedType.typeName.typeId
+    });
+}
+
+function convertExampleContainerTypeToTypeReference(
+    exampleContainerType: FernIr.ExampleTypeReferenceShape.Container
+): FernIr.TypeReference.Container {
+    switch (exampleContainerType.container.type) {
+        case "list":
+            return FernIr.TypeReference.container(FernIr.ContainerType.list(exampleContainerType.container.itemType));
+        case "set":
+            return FernIr.TypeReference.container(FernIr.ContainerType.set(exampleContainerType.container.itemType));
+        case "optional":
+            return FernIr.TypeReference.container(
+                FernIr.ContainerType.optional(exampleContainerType.container.valueType)
+            );
+        case "nullable":
+            return FernIr.TypeReference.container(
+                FernIr.ContainerType.nullable(exampleContainerType.container.valueType)
+            );
+        case "map":
+            return FernIr.TypeReference.container(
+                FernIr.ContainerType.map({
+                    keyType: exampleContainerType.container.keyType,
+                    valueType: exampleContainerType.container.valueType
+                })
+            );
+        case "literal":
+            return FernIr.TypeReference.container(
+                FernIr.ContainerType.literal(convertExampleLiteralToLiteral(exampleContainerType.container))
+            );
+        default:
+            assertNever(exampleContainerType.container);
+    }
+}
+
+function convertExampleLiteralToLiteral(literalContainer: FernIr.ExampleLiteralContainer): FernIr.Literal {
+    switch (literalContainer.literal.type) {
+        case "string":
+            return FernIr.Literal.string(literalContainer.literal.string.original);
+        case "boolean":
+            return FernIr.Literal.boolean(literalContainer.literal.boolean);
+        case "integer":
+        case "long":
+        case "uint":
+        case "uint64":
+        case "float":
+        case "double":
+        case "date":
+        case "datetime":
+        case "uuid":
+        case "base64":
+        case "bigInteger":
+            throw new Error("Not implemented");
```

Comment:
Although unlikely, we should add more context here - 
```suggestion
            throw new Error("Internal error; only boolean and string literals are permitted");
```